beast/physicsmodel/tests/test_stellar_prior_weights.py

```diff
@@ -1,186 +1,134 @@
 import numpy as np
 
-from beast.physicsmodel.prior_weights_stars import (
-    compute_distance_prior_weights,
-    compute_age_prior_weights,
-    compute_mass_prior_weights,
-    compute_metallicity_prior_weights,
-    imf_kroupa,
+from beast.physicsmodel.priormodel import (
+    PriorAgeModel,
+    PriorMassModel,
+    PriorMetallicityModel,
+    PriorDistanceModel,
 )
 
 
-def test_flat_age_prior_weights():
+def test_age_prior_weights():
     """
-    Test for flat age prior
+    Test the age prior weights
     """
+
     log_age = np.array([6.0, 7.0, 8.0, 9.0, 10.0])
-    log_age_prior_model = {"name": "flat", "sfr": 1.0}
-    log_age_prior = compute_age_prior_weights(log_age, log_age_prior_model)
-    expected_log_age_prior = [1, 1, 1, 1, 1]
-    np.testing.assert_allclose(
-        log_age_prior, expected_log_age_prior, err_msg=("Flat age prior error")
-    )
 
+    models = [
+        {"name": "flat"},
+        {"name": "flat", "amp": 0.1},
+        {"name": "flat_log"},
+        {
+            "name": "bins_histo",
+            "x": [6.0, 7.0, 8.0, 9.0, 10.0],
```

Comment:
can you do a test for a case with len(logages) = len(values)-1 for the bins_histo?

In the beast_settings, age_prior_model has the keyword 'logages' for age information. Thus, "x" -> "logages".

Reply:
Yes, I have added that test.  Good catch!